core/src/test/java/com/linecorp/armeria/server/healthcheck/DefaultCpuHealthCheckerTest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.healthcheck;
+
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.lang.reflect.Method;
+
+import org.junit.jupiter.api.Test;
+
+class DefaultCpuHealthCheckerTest {
```

Comment:
Could you improve this test so that:

- It actually checks the CPU usage and load (just some range check will be enough)
- It actually checks the healthiness is reported correctly.
  - You could refactor `DefaultCpuHealthChecker` to accept `Supplier`s that provide usage and load, and then inject mock suppliers to test this.

e.g.

```java
class CpuHealthChecker ... {
    CpuHealthChecker(...) {
        this(CpuHealthChecker::cpuUsage, CpuHealthChecker::cpuLoad, ...)
    }

    @VisibleForTesting
    CpuHealthChecker(DoubleSupplier cpuUsageSupplier, DoubleSupplier cpuLoadSupplier, ...) { ... }
}

// In your test:
@Test
void test() {
    final CpuHealthChecker checker = new CpuHealthChecker(() -> 3.0, () -> 4.0, ...);
    // Assertions here
}
```

Reply:
I've implemented your suggestion by creating a method for testing.
Please take a look and leave a review to see if I did what you suggested.🙏